homeassistant/components/modbus/__init__.py

```diff
@@ -87,13 +104,52 @@
     }
 )
 
+COVERS_SCHEMA = vol.All(
+    cv.has_at_least_one_key(CALL_TYPE_COIL, CONF_REGISTER),
+    vol.Schema(
+        {
+            vol.Required(CONF_HUB): cv.string,
+            vol.Required(CONF_NAME): cv.string,
+            vol.Optional(
+                CONF_SCAN_INTERVAL, default=DEFAULT_SCAN_INTERVAL
+            ): cv.positive_int,
```

Comment:
This should be `cv.time_period` as primary validator. Convert to seconds with a second validator.

```py
vol.All(cv.time_period, lambda value: value.total_seconds())
```